### PR TITLE
[pouchdb-adapter-indexeddb] Add types

### PR DESCRIPTION
**Pull request by cstrickland-WE**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

# Notes

- Structured based off of other pouchdb-adapter-* types

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/pouchdb-adapter-indexeddb/pouchdb-adapter-indexeddb-tests.ts
1. types/pouchdb-adapter-indexeddb/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ac68179b6afd81cc7786ccaa6f0266116525d300
</p>

**Review by gabritto — DISMISSED** [review]
Technically, this module exports an index function (which you then pass in to `PouchDB`, according to the package readme). Please add that.

**Comment by cstrickland-WE** [comment]
> Technically, this module exports an index function (which you then pass in to `PouchDB`, according to the package readme). Please add that.

Made a change based on your comment. Let me know if that is incorrect.

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/pouchdb-adapter-indexeddb/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 63addeb0e7eb033c8adf5635f81eed0d94a1b6e7
</p>

File: types/pouchdb-adapter-indexeddb/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            name: string | null,
+            options: IndexedDBAdapter.IndexedDBAdapterConfiguration,
+        ): Database<Content>;
+    }
+}
+
+// eslint-disable-next-line @definitelytyped/no-single-declare-module, @definitelytyped/no-declare-current-package
+declare module "pouchdb-adapter-indexeddb" {
+    export default function (PouchDB: PouchDB.Static): void;
+}
```
Comment (andrewbranch) [inline types/pouchdb-adapter-indexeddb/index.d.ts:51]: Please don’t disable lint rules—copy https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/pouchdb-adapter-node-websql/index.d.ts as an example.

**Review by andrewbranch — CHANGES_REQUESTED** [review]

**Review by andrewbranch — APPROVED** [review]
Thanks!

**Comment by cstrickland-WE** [comment]
Ready to merge

**Pull request merged**